aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -103,14 +105,30 @@ protected void doRefresh() {
   @Override
   protected void doCommit(TableMetadata base, TableMetadata metadata) {
     String newMetadataLocation = writeNewMetadata(metadata, currentVersion() + 1);
-    boolean exceptionThrown = true;
+    CommitStatus commitStatus = CommitStatus.FAILURE;
+
     try {
       lock(newMetadataLocation);
       Table glueTable = getGlueTable();
       checkMetadataLocation(glueTable, base);
       Map<String, String> properties = prepareProperties(glueTable, newMetadataLocation);
-      persistGlueTable(glueTable, properties);
-      exceptionThrown = false;
+
+      try {
+        persistGlueTable(glueTable, properties);
+        commitStatus = CommitStatus.SUCCESS;
+      } catch (Throwable persistFailure) {
```

Comment:
1. I think this can be `SdkException` instead of all throwables, because `persistGlueTable` is only a call to Glue. If it is not that exception type, it is guaranteed to be a failure.
2. can we avoid nested try and directly determine this in the catch block below? We are already catching that exception at L137 anyway.

Reply:
Thanks for the feedback! 

For 1, I'm not confident if any type of network issue would end up as `SdkException`, so I'd prefer to maintain a list of exceptions that won't go through `checkCommitStatus` than a list that we will just to be on the safe side. 
For 2, sounds good, I'll move the catch logic to L137 but I'll convert it to a `Throwable` to handle 1. This essentially means to not check commit status for `ConcurrentModificationException` and `AlreadyExistsException` which I think should be minimum risk.

Reply:
catching `Throwable` always sound very dangerous to me, because `Error` indicates serious problems that a reasonable application should not try to catch,  and I would much prefer just catching `RuntimeException` or at least just `Exception`. Was there any particular reason for `HiveTableOperations` to catch `Throwable`? @RussellSpitzer

Reply:
I think it is less important now that we start out with the commit state set to unknown, but in the original design we basically started out assuming that the commit had succeeded and switched a flag to indicate that it had failed. We can probably have this just be runtime exception now since now the logic is basically for deciding when to retry.

Reply:
Sounds good, I was thinking if it's a `Error` we will very unlikely be able have a success commit before that, so it will be rethrown anyway, but on the other hand we probably want to make sure the process to die immediately. I'll update L137 to catch runtime exception.